frontend/src/_ui/Search/Search.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+export const SearchBox = ({ onChange, ...restProps }) => {
+  const { callback, placeholder } = restProps;
+  const [searchText, setSearchText] = React.useState('');
+
+  const handleChange = (e) => {
+    setSearchText(e.target.value);
+    onChange(e.target.value);
+  };
+
+  const clearSearch = () => {
+    setSearchText('');
+    onChange('');
+  };
+
+  React.useEffect(() => {
+    if (searchText) {
+      document.querySelector('.searchbox-wrapper .input-icon .form-control:not(:first-child)').style.paddingLeft =
+        '0.5rem';
+    }
+
+    return () => {
+      document.querySelector('.searchbox-wrapper .input-icon .form-control:not(:first-child)').style.paddingLeft =
+        '2.5rem';
+    };
+  }, [searchText]);
+
+  const componentWillUnmount = React.useRef(false);
+
+  // componentWillUnmount
+  React.useEffect(() => {
+    return () => {
+      componentWillUnmount.current = true;
+    };
+  }, []);
+
+  React.useEffect(() => {
+    return () => {
+      if (componentWillUnmount.current) {
+        setSearchText('');
+      }
+    };
+  }, [searchText]);
```

Comment:
If component unmounts and mounts again the `searchText` would be empty I assume, why is this required?

Reply:
similar bug to modal. but fixed that yesterday so we don't need that side effect to clear the fields on mounting and unmounting too